### ISSUE TITLE
Error: error creating SSH agent: "SSH agent requested but SSH_AUTH_SOCK not-specified" on `git bug push`

### ISSUE THREAD

**Issue by NobbZ** [MichaelMure/git-bug#1023]
Please guide me to a way to provide additional information…

Yes, I use SSH URLs for git.
No, they are not password protected.

If though starting SSH-Agent *and* adding the key, the error becomes:

> Error: ssh: handshake failed: knownhosts: key mismatch

Regular git operations against the remote work.

**Comment by patricioantoniotoledo** [comment]
cd ~/.ssh
ssh-add ~/.ssh/id_rsa

might help

**Comment by NobbZ** [comment]
To be honest, I do not even (want to) use SSH-agent.

I want the configured SSH-config to be respected and the command to just work.

I stoped using `git-bug` because of this a couple of days after posting this issue without a reply.

**Comment by octvs** [comment]
Might be [related](https://github.com/MichaelMure/git-bug/issues/1109#issuecomment-1732362208), I had some similar issues with `pull` & `push`, building from source fixed.

**Comment by stackengineer** [comment]
Not stale. Also looking for an agentless solution, preferably respecting `~/.ssh/config`.

**Comment by sudoforge** [comment]
> git's shorthand (scp like syntax) won't work for me (until ~/.ssh/config gets parsed) as I need to set a custom port. The scp like syntax does not allow setting a custom port number.

yeah, this is something i know about -- parsing custom ssh config is on my radar.

in the meantime, if your only need is to specify a custom port, using the full RFC-3986-compliant URI would work for you:

```
ssh://you@host:1234/foo/bar.git
```

> A SSH Agent is still needed. If I try a git bug pull or push without starting an agent, I get:
>
> ```
> Error: error creating SSH agent: "SSH agent requested but SSH_AUTH_SOCK not-specified"
> ```

yep - this is on my radar as well.

generally speaking, the way git-bug is currently written makes a lot of assumptions about your git configuration, your ssh configuration, and the types of remotes you're interacting with. these are the core issues i want to solve in the near term, as they are silly, and outright blocking people.
